cozy/codegen/optimization.py

```diff
@@ -368,9 +368,6 @@ def visit_EUnaryOp(self, e):
                 SForEach(loop_var, e.e,
                     SAssign(sum_var, EBinOp(sum_var, "+", loop_var).with_type(INT)))])))
             return sum_var
-        elif op == UOp.Length:
-            arg = EVar("x").with_type(e.e.type.elem_type)
-            return self.visit(EUnaryOp(UOp.Sum, EMap(e.e, ELambda(arg, ONE)).with_type(INT_BAG)).with_type(INT))
```

Comment:
You should also update the docstring on `simplify_and_optimize` so that it no longer claims to eliminate the `Length` operator.